pulp_rpm/app/shared_utils.py

```diff
@@ -16,28 +16,24 @@ def _prepare_package(artifact, filename):
     Helper function for creating package.
 
     Copy file to a temp directory under
-    the user provided filename or constructed N(e)VRA
-    from package info.
+    the user provided filename if provided.
 
-    Returns: artifact model as dict
+    Returns: package model as dict
 
     Args:
         artifact: inited and validated artifact to save
         filename: name of file uploaded by user
     """
-    artifact_file = storage.open(artifact.file.name)
-    cr_pkginfo = createrepo_c.package_from_rpm(
-        artifact_file.name, changelog_limit=settings.KEEP_CHANGELOG_LIMIT
-    )
-
     if not filename:
-        filename = (
-            f"{cr_pkginfo.name}-{cr_pkginfo.version}-{cr_pkginfo.release}.{cr_pkginfo.arch}.rpm"
-        )
+        filename = f"{artifact.pulp_id}.rpm"
```

Comment:
Do you mean to not take the name at all? (to remove a method's argument)

Reply:
I mean that this is the copy we're doing only so that createrepo_c parse the file, and so long as createrepo_c can parse the file the filename probably doesn't matter.